src/app/_hooks/useSearch.ts

```diff
@@ -0,0 +1,36 @@
+import { useMemo } from 'react'
+
+import { NextServerSearchParams } from '@/types/searchParams'
+import { SearchableByTitle } from '@/types/searchTypes'
+
+import { SEARCH_KEY } from '@/constants/searchParams'
+
+function validateSearchQuery(searchQuery: NextServerSearchParams['search']) {
```

Comment:
Can we have a look at `validateSortOptionQuery` - and see if we can abstract some of the logic?

Reply:
Hmm, maybe we can abstract the following? Not sure if it's adding a lot of value.

```ts
const rawQuery = Array.isArray(searchQuery) ? searchQuery[0] : searchQuery
return rawQuery.toLowerCase()
```

Reply:
What about something like `queryUtils.ts` - This is a rough draft - We can create types instead of having `key: string` etc.

```
const getFirstQueryParam = (params: NextServerSearchParams, key: string): string | undefined => {
  const param = params[key];
  return Array.isArray(param) ? param[0] : param;
};

const normalizeQueryParam = (params: NextServerSearchParams, key: string): string | undefined => {
  const queryValue = getFirstQueryParam(params, key);
  return queryValue ? queryValue.toLowerCase() : undefined;
};

```

I'm saying this because we actually also use this for `usePagination`

```
  if (Array.isArray(pageQuery)) {
    currentPage = validatePageNumber(pageQuery[0], pageCount)
  } else {
    currentPage = validatePageNumber(pageQuery, pageCount)
  }
```

And we will be using it categories as well, I assume. That means, at least, 4 places for now. Seems worth abstracting?

Reply:
✅